src/pymmcore_plus/core/_mmcore_plus.py

```diff
@@ -1734,24 +1734,30 @@ def saveSystemConfiguration(self, filename: str) -> None:
         **Why Override?** To also save pixel size configurations.
         """
         super().saveSystemConfiguration(filename)
-        px_configs = self.getAvailablePixelSizeConfigs()
-        if not px_configs:
-            return
         # saveSystemConfiguration does not save the pixel size config so here
         # we add to the saved file also any pixel size config.
+        self._save_pixel_configurations(filename)
+
+    def _save_pixel_configurations(self, filename: str) -> None:
         with open(filename, "a") as f:

```

Comment:
:joy: you moved it here, but _still_ opened the file without checking if there are any configs that need to be saved! :)

move the open to where it's needed: just immediately before you write to the file.  line 1760

Reply:
think of it like opening the refrigerator: you don't open the fridge, then go make coffee, then come back to the fridge to pull out the eggs :)

Reply:
yup! make sense :)